File: app/src/main/java/com/example/foodrecipes/requests/ServiceGenerator.java
package com.example.foodrecipes.requests;

import com.example.foodrecipes.util.Constants;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ServiceGenerator {

    private static Retrofit.Builder retrofitBuilderForInfo =
            new Retrofit.Builder()
            .baseUrl(Constants.BASE_URL_INFO)
            .addConverterFactory(GsonConverterFactory.create());

    private static Retrofit retrofitForInfo = retrofitBuilderForInfo.build();

    private static InfoApi infoApi = retrofitForInfo.create(InfoApi.class);

    public static InfoApi getInfoApi() {
        return infoApi;
    }

    private static Retrofit.Builder retrofitBuilderForImage =
            new Retrofit.Builder()
            .baseUrl(Constants.BASE_URL_IMAGE)
            .addConverterFactory(GsonConverterFactory.create());

    private static Retrofit retrofitForImage = retrofitBuilderForImage.build();

    private static ImageApi imageApi = retrofitForImage.create(ImageApi.class);

    public static ImageApi getImageApi(){
        return imageApi;
    }
}

File: app/src/main/java/com/example/foodrecipes/requests/ServiceGenerator.kt
package com.example.foodrecipes.requests

import com.example.foodrecipes.util.Constants
import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ServiceGenerator {

    private val retrofitBuilderForInfo = Retrofit.Builder()
            .baseUrl(Constants.BASE_URL_INFO)
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(CoroutineCallAdapterFactory())

    private val retrofitForInfo = retrofitBuilderForInfo.build()

    val infoApi = retrofitForInfo.create(InfoApi::class.java)

    private val retrofitBuilderForImage = Retrofit.Builder()
            .baseUrl(Constants.BASE_URL_IMAGE)
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(CoroutineCallAdapterFactory())

    private val retrofitForImage = retrofitBuilderForImage.build()

    val imageApi = retrofitForImage.create(ImageApi::class.java)
}